tests/pairedomics/test_strain_mappings_generator.py

```diff
@@ -0,0 +1,288 @@
+import json
+from nplinker.pairedomics import extract_mappings_ms_filename_spectrum_id
+from nplinker.pairedomics import \
+    extract_mappings_original_genome_id_resolved_genome_id
+from nplinker.pairedomics import extract_mappings_resolved_genome_id_bgc_id
+from nplinker.pairedomics import extract_mappings_strain_id_ms_filename
+from nplinker.pairedomics import extract_mappings_strain_id_original_genome_id
+from nplinker.pairedomics import get_mappings_strain_id_bgc_id
+from nplinker.pairedomics import get_mappings_strain_id_spectrum_id
+from nplinker.pairedomics import podp_generate_strain_mappings
+from nplinker.strain_collection import StrainCollection
+from nplinker.strains import Strain
+
+
+def test_podp_generate_strain_mappings(monkeypatch, tmp_path):
+    # mock functions called by the tested function
+    mappings_strain_bgc = {
+        "strain1": {"bgc1", "bgc2"},
+        "strain2": {"bgc3"},
+    }
+    mappings_strain_spectrum = {
+        "strain1": {"spec1", "spec2"},
+        "strain2": {"spec3"}
+    }
+
+    monkeypatch.setattr(
+        "nplinker.pairedomics.strain_mappings_generator.extract_mappings_strain_id_original_genome_id",
+        lambda *args: {})  # any return value is fine
+    monkeypatch.setattr(
+        'nplinker.pairedomics.strain_mappings_generator.extract_mappings_original_genome_id_resolved_genome_id',
+        lambda *args: {})
+    monkeypatch.setattr(
+        'nplinker.pairedomics.strain_mappings_generator.extract_mappings_resolved_genome_id_bgc_id',
+        lambda *args: {})
+    monkeypatch.setattr(
+        "nplinker.pairedomics.strain_mappings_generator.get_mappings_strain_id_bgc_id",
+        lambda *args: mappings_strain_bgc)
+
+    monkeypatch.setattr(
+        'nplinker.pairedomics.strain_mappings_generator.extract_mappings_strain_id_ms_filename',
+        lambda *args: {})
+    monkeypatch.setattr(
+        'nplinker.pairedomics.strain_mappings_generator.extract_mappings_ms_filename_spectrum_id',
+        lambda *args: {})
+    monkeypatch.setattr(
+        "nplinker.pairedomics.strain_mappings_generator.get_mappings_strain_id_spectrum_id",
+        lambda *args: mappings_strain_spectrum)
+
+    # Create the expected
+    expected_dict = {
```

Comment:
Why not using `@pytest.mark.parametrize` for the expected results? Also in the other tests

Reply:
`@pytest.mark.parametrize` is usually used when we have multiple types of input for testing. Only one type of input is considered in these unit tests (it's reasonable at the moment), so that's why `@pytest.mark.parametrize` is not used.

**Related issue:**
The input data (e.g. table data, json data) should be validated with a framework. I'm going to work on that in next PR.